sdk/storage/azure-storage-files-datalake/test/ut/datalake_sas_test.cpp

```diff
@@ -207,271 +327,480 @@ namespace Azure { namespace Storage { namespace Test {
+             Sas::DataLakeSasPermissions::ManageAccessControl,
          })
     {
-      filesystemSasBuilder.SetPermissions(permissions);
-      auto sasToken = filesystemSasBuilder.GenerateSasToken(*keyCredential);
-      auto sasToken2 = filesystemSasBuilder.GenerateSasToken(userDelegationKey, accountName);
-
-      if ((permissions & Sas::DataLakeFileSystemSasPermissions::All)
-          == Sas::DataLakeFileSystemSasPermissions::All)
+      directorySasBuilder.SetPermissions(permissions);
+      auto sasToken2 = directorySasBuilder.GenerateSasToken(userDelegationKey, accountName);
+      if ((permissions & Sas::DataLakeSasPermissions::Read) == Sas::DataLakeSasPermissions::Read)
       {
-        (void)verify_file_move;
-        /*
-        don't know why, move doesn't work
-        verify_file_move(sasToken);
-        verify_file_move(sasToken2);
-        */
+        VerifyDataLakeSasRead(dataLakeDirectoryClient, sasToken2);
       }
-      if ((permissions & Sas::DataLakeFileSystemSasPermissions::Read)
-          == Sas::DataLakeFileSystemSasPermissions::Read)
+      if ((permissions & Sas::DataLakeSasPermissions::Delete)
+          == Sas::DataLakeSasPermissions::Delete)
       {
-        verify_file_read(sasToken);
-        verify_file_read(sasToken2);
+        VerifyDataLakeSasDelete(dataLakeDirectoryClient, sasToken2);
       }
-      if ((permissions & Sas::DataLakeFileSystemSasPermissions::Write)
-          == Sas::DataLakeFileSystemSasPermissions::Write)
+      if ((permissions & Sas::DataLakeSasPermissions::List) == Sas::DataLakeSasPermissions::List)
       {
-        verify_file_write(sasToken);
-        verify_file_write(sasToken2);
+        VerifyDataLakeSasList(dataLakeDirectoryClient, sasToken2);
       }
-      if ((permissions & Sas::DataLakeFileSystemSasPermissions::Delete)
-          == Sas::DataLakeFileSystemSasPermissions::Delete)
+      if ((permissions & Sas::DataLakeSasPermissions::Add) == Sas::DataLakeSasPermissions::Add)
       {
-        verify_file_delete(sasToken);
-        verify_file_delete(sasToken2);
+        /*
+         * Add test for append block when DataLake supports append blobs.
+         */
       }
-      if ((permissions & Sas::DataLakeFileSystemSasPermissions::List)
-          == Sas::DataLakeFileSystemSasPermissions::List)
+      if ((permissions & Sas::DataLakeSasPermissions::Create)
+          == Sas::DataLakeSasPermissions::Create)
       {
-        verify_filesystem_list(sasToken);
-        verify_filesystem_list(sasToken2);
+        VerifyDataLakeSasCreate(dataLakeDirectoryClient, sasToken2);
       }
-      if ((permissions & Sas::DataLakeFileSystemSasPermissions::Add)
-          == Sas::DataLakeFileSystemSasPermissions::Add)
+      if ((permissions & Sas::DataLakeSasPermissions::Move) == Sas::DataLakeSasPermissions::Move)
       {
-        verify_file_add(sasToken);
-        verify_file_add(sasToken2);
+        // Don't know why, move doesn' t work
```

Comment:
Does it still not work?

If not, need to track it in https://github.com/Azure/azure-sdk-for-cpp/issues/4850

Reply:
Yeah, It can work when SasResourceType is FileSystem and can't work when SasResourceType is Directory.

Reply:
If this is already covered in filesystem sas test, I think we can remove this one, instead of leaving a mysterious commen here.

Reply:
Ok. I will remove this in the next PR.